### ISSUE TITLE
Feature request - API secret/authentication

### ISSUE THREAD

**Issue by hidden-69** [UnamSanctam/UnamWebPanel#100]
Pass a secret in your miner build to connect to the webpanel/api, if it doesn't match, don't connect.

This way no one can get away with hijacking your domain/tunnel and setup an new WebPanel.

**Comment by UnamSanctam** [comment]
Does it matter? How would someone hijack your domain?

**Comment by hidden-69** [comment]
When you use ngrok to setup a tunnel and make your webpanel publicly available, someone else will get your fqdn after you close the tunnel.

**Comment by UnamSanctam** [comment]
But wouldn't you just lose your web panel anyways then? Wouldn't it be better to just not use the web panel if you're inevitably going to lose it? And why wouldn't use you use a fixed domain that you then redirect to the fqdn if you're going to use ngrok?

**Comment by hidden-69** [comment]
ngrok doesn't work that way, you have to use the fqdn you get or your tunnel won't work. The server probably does some forwarding based on hostname.

You can replace the url in your remote config when you setup a new tunnel :) it's a free and secure way to set it up like this.

**Comment by UnamSanctam** [comment]
Well it's not really free since you still either need a VPN or VPS to host it on/behind (unless you got that for free of course) but I don't think encrypting the JSON (which would be the only realistic solution to authentication due to how it works) is really needed.

**Comment by hidden-69** [comment]
with ngrok you can host in on your laptop on any random network without incoming port, you get a fqdn, change it in your remote config, and your clients connect after they update.

What I want to prevent, is that after a couple of days if I close ngrok, the clients connect with the old config to the same tunnel that someone else uses. Could also change the config and remove the API...

**Comment by UnamSanctam** [comment]
Yes sure just meant that you'd need a VPS/VPN if you don't want to expose your IP to ngrok, but if you're not on your own internet I guess that wouldn't matter. What's the chance of someone else getting the same fqdn as you had and then setting up their own web panel? And I guess that even if someone got your fqdn they would still lose the miner from their web panel after the miners next restart since it would read the new API URL from your "Remote Configuration" (because with your configuration you wouldn't need any API URL inside the builder).

**Comment by hidden-69** [comment]
That's a real chance since it uses numbers and assigns lowest available, 0.ngrok, for example.

Not saying it will happen regularly, it's something that could be a security risk, some host validation would be a great feature.
